vm_insnhelper.c

```diff
@@ -4646,6 +4646,9 @@ vm_search_super_method(const rb_control_frame_t *reg_cfp, struct rb_call_data *c
 
     ID mid = me->def->original_id;
 
+    if (!vm_ci_packed_p(cd->ci))
+        cd->ci->kwarg->references++;
```

Comment:
I don't think we need to increment `references` here.

Reply:
Thanks! I think this improved the leak situation! It still takes a while to level off, but it is improved.